packages/dd-trace/src/plugins/util/test.js

```diff
@@ -116,6 +118,35 @@ function getPkgManager () {
   }
 }
 
+function validateMetadata (metadata) {
+  if (GIT_REPOSITORY_URL in metadata && metadata[GIT_REPOSITORY_URL]) {
+    const validUrl = validateUrl(metadata[GIT_REPOSITORY_URL])
+    if (!validUrl) {
+      delete metadata[GIT_REPOSITORY_URL]
+    }
+  }
+  if (CI_PIPELINE_URL in metadata && metadata[CI_PIPELINE_URL]) {
+    const validUrl = validateUrl(metadata[CI_PIPELINE_URL])
+    if (!validUrl) {
+      delete metadata[CI_PIPELINE_URL]
+    }
+  }
+  return removeEmptyValues(metadata)
+}
+
+function validateUrl (url) {
+  if (url.startsWith('git@') && url.includes(':')) {
+    const hostname = url.substring(url.indexOf('@') + 1, url.lastIndexOf(':'))
```

Comment:
maybe a more robust way of testing this type of URLs is using a regex like https://gist.github.com/franciskim/41a959f8e3989254ef5d (TBD if this works)